packages/chopsticks/src/plugins/follow-chain/index.ts

```diff
@@ -0,0 +1,50 @@
+import { ApiPromise, HttpProvider, WsProvider } from '@polkadot/api'
+import { ProviderInterface } from '@polkadot/rpc-provider/types'
+import { defaultLogger } from '@acala-network/chopsticks-core'
+import _ from 'lodash'
+import type yargs from 'yargs'
+
+import { Config } from '../../schema'
+import { defaultOptions } from '../../cli-options'
+import { setupContext } from '../../context'
+
+const options = _.pick(defaultOptions, ['endpoint', 'wasm-override', 'runtime-log-level', 'offchain-worker'])
+const logger = defaultLogger.child({ name: 'follow-chain' })
+
+export const cli = (y: yargs.Argv) => {
+  y.command(
+    'follow-chain',
+    'Always follow the latest block on upstream',
+    (yargs) => yargs.options(options),
+    async (argv) => {
+      const { chain } = await setupContext(argv as Config, true)
+
+      let provider: ProviderInterface
+      const endpoint = argv.endpoint as string
+      if (/^(https|http):\/\//.test(endpoint || '')) {
+        provider = new HttpProvider(endpoint as string)
+      } else {
+        provider = new WsProvider(endpoint as string)
+      }
+      const apiPromise = await ApiPromise.create({
```

Comment:
using an ApiPromise here to subscribe, is there a another way? like adding method in chain.api

Reply:
another way is subscribe directly from the WsProvider

Reply:
yeah try do avoid creating ApiPromise. It's better to create an api method that does the subscription